### PR TITLE
Introduce RCT2 namespace

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Using this to help identify things that shouldn't be here.

### CODE REVIEW

**Review by Gymnasiast — APPROVED** [review]

**Review by Broxzier — APPROVED** [review]

File: src/openrct2/rct1/S4Importer.cpp [hunk 1]
```diff
@@ -2499,7 +2499,7 @@ namespace RCT1
         {
             const auto originalString = _s4.string_table[(stringId - USER_STRING_START) % 1024];
             auto originalStringView = std::string_view(
-                originalString, GetRCT2StringBufferLen(originalString, USER_STRING_MAX_LENGTH));
```
Comment (duncanspumpkin) [inline src/openrct2/rct1/S4Importer.cpp:2502]: Clearly this isn't an RCT2 function but an RCT12 function to be moved in future pr.

File: src/openrct2/management/Research.cpp [hunk 2]
```diff
@@ -83,7 +83,7 @@ ResearchItem::ResearchItem(const RCT12ResearchItem& oldResearchItem)
         if (type == Research::EntryType::Ride)
         {
             auto* rideEntry = get_ride_entry(entryIndex);
-            baseRideType = rideEntry != nullptr ? RCT2RideTypeToOpenRCT2RideType(oldResearchItem.baseRideType, rideEntry)
+            baseRideType = rideEntry != nullptr ? RCT2::RCT2RideTypeToOpenRCT2RideType(oldResearchItem.baseRideType, rideEntry)
```
Comment (duncanspumpkin) [inline src/openrct2/management/Research.cpp:86]: Import to be moved https://github.com/OpenRCT2/OpenRCT2/issues/16013

File: src/openrct2/management/Research.h [hunk 3]
```diff
@@ -97,7 +97,7 @@ struct ResearchItem
         else
         {
             retItem.entryIndex = OpenRCT2EntryIndexToRCTEntryIndex(entryIndex);
-            retItem.baseRideType = OpenRCT2RideTypeToRCT2RideType(baseRideType);
+            retItem.baseRideType = RCT2::OpenRCT2RideTypeToRCT2RideType(baseRideType);
```
Comment (duncanspumpkin) [inline src/openrct2/management/Research.h:100]: To be deleted. Function unused https://github.com/OpenRCT2/OpenRCT2/issues/16013

File: src/openrct2/scenario/ScenarioRepository.cpp [hunk 4]
```diff
@@ -269,7 +269,7 @@ class ScenarioFileIndex final : public FileIndex<scenario_index_entry>
         return false;
     }
 
-    static scenario_index_entry CreateNewScenarioEntry(const std::string& path, uint64_t timestamp, rct_s6_info* s6Info)
+    static scenario_index_entry CreateNewScenarioEntry(const std::string& path, uint64_t timestamp, RCT2::S6Info* s6Info)
```
Comment (duncanspumpkin) [inline src/openrct2/scenario/ScenarioRepository.cpp:272]: This looks like import code. Move to RCT2 folder.

File: src/openrct2/Game.cpp [hunk 5]
```diff
@@ -385,7 +385,7 @@ void game_convert_strings_to_utf8()
 /**
  * Converts all the user strings and news item strings to RCT2 encoding.
  */
-void game_convert_strings_to_rct2(rct_s6_data* s6)
+void game_convert_strings_to_rct2(RCT2::S6Data* s6)
```
Comment (duncanspumpkin) [inline src/openrct2/Game.cpp:388]: These are import functions and should be moved.

**Comment by duncanspumpkin** [comment]
My comments will be addressed in follow on PRs. Also need to perform this same namespacing to the RCT12 header.

**Pull request merged**